keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,465 @@
+---
+kep-number: draft-20180426
+title: Security Conformance
+authors:
+  - "@easeway"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-testing
+  - sig-api-machinery
+  - sig-network
+  - sig-node
+  - sig-cluster-lifecycle
+reviewers:
+  - "@tallclair"
+  - "@ericchiang"
+  - "@liggitt"
+  - "@davidopp"
+approvers:
+  - "@tallclair"
+  - "@ericchiang"
+  - "@liggitt"
+editor: "@easeway"
+creation-date: 2018-04-26
+status: provisional
+---
+
+# Security Conformance
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Security Conformance Profiles](#security-conformance-profiles)
+    * [Behavioral Criteria](#behavioral-criteria)
+    * [Non-behavioral Criteria](#non-behavioral-criteria)
+    * [Implementation Details](#implementation-details)
+* [Graduation Criteria](#graduation-criteria)
+
+## Summary
+
+Security Conformance defines how a Kubernetes cluster will be considered as secure,
+and also proposes the mechanisms to configure a cluster securely.
+
+## Motivation
+
+Configuring a Kubernetes cluster is complicated,
+and it's very easy to mis-configure a cluster to be insecure.
+The current Kubernetes conformance test suite barely covers security aspects.
+
+### Goals
+
+- A list of behavioral security criteria that can be validated by tests
+- A list of non-behavioral security criteria that can be validated by tests
+
+### Non-Goals
+
+- Mechanisms to satisfy security criteria
+- Non-security related
+- Details about Kubernetes conformance test
+
+## Proposal
+
+### Security Conformance Profiles
+
+The behavioral/non-behavioral criteria are categorized into a few security conformance profiles for different levels of security requirements:
+
+- `Essential`: the baseline security requirements for the cluster,
```

Comment:
The three definitions of _Security Conformance Profiles_ are based on where the threat comes from, and who will be impacted. 

Generally, the threat comes from:

1. Outside of the cluster: the attack surface will be K8S APIs, the hosts including master and nodes.
2. Inside the cluster: the workload running on K8S cluster

The impact will be on:
1. The cluster: the cluster may not function or leaking information
2. The workload: the cluster is still functioning, and secure, while some workload will be compromised, and may compromise other workload

_Essential_ is actually defined for _trusted_ workloads: it focuses on the threat from outside of the cluster and impact on the cluster. It will assume the workload is trusted, and allow privileges on workload. The common use case is a K8S cluster used internally in a team, or an org with all trusted users.

_Restricted_ is based on _Essential_, and additionally targets the threat from inside. The targeted impact is still on the cluster, not workloads. The common use case will be for untrusted workloads, regardless of tenants. For example, the operators of K8S cluster is responsible for keeping the cluster secure and stable, not considering the influence between workloads.

_Isolated_ based on _Restricted_, targets threat from both inside and outside, impact on both cluster and workload. It's for multi-tenancy use cases. It will prevent the attack from some tenants' workload to other tenants.

I will give more details on the definitions - mentioning use case categories.